### ISSUE TITLE
`/etc/mysql/my.cnf` is not installed

### ISSUE THREAD

**Issue by gasuketsu** [debops/ansible-mariadb_server#50]
I found that `/etc/mysql/my.cnf` is not installed during play, and this causes no config files under `/etc/mysql/conf.d` are loaded when starting mysql/mariadb server.
Current playbook creates `zz-mysqld.cnf`, so it looks that no configuration/option is applied to server.
(I have tested with Ubuntu 14.04 box)

My question is, the top-level config file `/etc/mysql/my.cnf' outside of play manually ?

If current situation is not expected and all configuration/option should be applied automatically, I think we just need to `my.cnf` that just include files under `/etc/mysql/conf.d` like below.

```
# include all conf files (both debops managed and package provides)
!includedir /etc/mysql/conf.d
```

or 

```
# just include debops managed
!include /etc/mysql/conf.d/zz-mysqld.cnf
!include /etc/mysql/conf.d/zz-client.cnf
```

**Comment by drybjed** [comment]
A quick check on my Debian host reveals that the `/etc/mysql/my.cnf` file is part of the `mysql-common` APT package:
```
# dpkg -S /etc/mysql/my.cnf 
mysql-common: /etc/mysql/my.cnf
# dpkg-query -s mysql-common
Package: mysql-common
Status: install ok installed
Priority: optional
Section: database
Installed-Size: 141
Maintainer: Debian MySQL Maintainers <pkg-mysql-maint@lists.alioth.debian.org>
Architecture: all
Multi-Arch: foreign
Source: mysql-5.5
Version: 5.5.57-0+deb8u1
Breaks: amarok (<< 2.5.0-2), mysql-client-5.1 (<< 5.5), mysql-server-5.1 (<< 5.5), mysql-server-core-5.1 (<< 5.5)
Conffiles:
 /etc/mysql/conf.d/.keepme d41d8cd98f00b204e9800998ecf8427e
 /etc/mysql/my.cnf e252af9e23d14b0215260ee023959f5a
Description: MySQL database common files, e.g. /etc/mysql/my.cnf
 MySQL is a fast, stable and true multi-user, multi-threaded SQL database
 server. SQL (Structured Query Language) is the most popular database query
 language in the world. The main goals of MySQL are speed, robustness and
 ease of use.
 .
 This package includes files needed by all versions of the client library,
 e.g. /etc/mysql/my.cnf.
Homepage: http://dev.mysql.com/
```
This usually means that the file is created during package installation. Since it's a Conffile, it's not directly managed by DebOps.

Can you check if you have the `mysql-common` package installed on Ubuntu?

**Comment by gasuketsu** [comment]
Thanks for quick reply.

I have verified that mysql-common package as one of dependencies.

I remember that I have tried running playbook several times with configuration/inventory changes, and purged mariadb-server package and whole fils under `/etc/mysql` before re-running.
(I have also performed `dpkg --list |grep "^rc"  | cut -d " " -f 3 | xargs sudo dpkg --purge` to remove whole configuration files provided by removed packages, but it seems that mysql-common was remained)

So, losing `my.cnf` in my box should be my fault.

Thanks.
